compiler/utils.go

```diff
@@ -650,5 +650,8 @@ func (s varsByName) Swap(i, j int) {
 }
 
 func (s varsByName) Less(i, j int) bool {
-	return s[i].Name() < s[j].Name()
+	if s[i].Name() != s[j].Name() {
+		return s[i].Name() < s[j].Name()
+	}
+	return s[i].Pos() < s[j].Pos()
```

Comment:
More often than not, I think names will differ. Only when they don't, do you want to sort by pos. Therefore, I think it'd be more readable to make the less frequent case indented, by flipping them:

```Go
if s[i].Name() == s[j].Name() {
	return s[i].Pos() < s[j].Pos()
}
return s[i].Name() < s[j].Name()
```

This is equivalent. Just a style suggestion.